From 4eb13cdcd2098ad79d4b93004d2cc4627fa0c863 Mon Sep 17 00:00:00 2001
From: datauwu <209150516+datauwu@users.noreply.github.com>
Date: Sat, 10 Oct 2026 06:31:18 +0800
Subject: [PATCH] archive_read: handle failed filename seeks

Return ARCHIVE_FAILED for failed lseek calls and range checks.
Let the reader track prior movement before trying other bidders.
---
 libarchive/archive_read_open_filename.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/libarchive/archive_read_open_filename.c b/libarchive/archive_read_open_filename.c
index 91f64b694a0c..2c442eb12050 100644
--- a/libarchive/archive_read_open_filename.c
+++ b/libarchive/archive_read_open_filename.c
@@ -583,7 +583,6 @@ file_seek(struct archive *a, void *client_data, int64_t request, int whence)
 	if (r >= 0)
 		return r;
 
-	/* If the input is corrupted or truncated, fail. */
 err:
 	if (mine->filename_type == FNT_STDIN)
 		archive_set_error(a, errno, "Error seeking in stdin");
@@ -593,7 +592,7 @@ err:
 	else
 		archive_set_error(a, errno, "Error seeking in '%ls'",
 		    mine->filename.w);
-	return (ARCHIVE_FATAL);
+	return (ARCHIVE_FAILED);
 }
 
 static int
